crates/runtime/src/flight/mod.rs

```diff
@@ -466,13 +470,25 @@ pub async fn start(
             .context(UnableToConfigureTlsSnafu)?;
     }
 
+    #[cfg(feature = "cluster")]
+    if tls_config.is_none()
```

Comment:
If starting in cluster mode without a TLS configuration, we panic and show the user this message. They have to acknowledge and pass an extra flag to run the flight service without TLS.